prow/spyglass/lenses/junit/lens.go

```diff
@@ -136,48 +178,78 @@ func (lens Lens) Body(artifacts []lenses.Artifact, resourceDir string, data stri
 	}
 	sort.Slice(results, func(i, j int) bool { return results[i].path < results[j].path })
 
-	jvd := struct {
-		NumTests int
-		Passed   []TestResult
-		Failed   []TestResult
-		Skipped  []TestResult
-	}{}
+	jvd := JVD{}
+
 	for _, result := range results {
 		if result.err != nil {
 			continue
 		}
-		for _, test := range result.junit {
-			if test.Failure != nil {
+		for _, tests := range result.junit {
+			var skipped, passed, failed, flaky bool
+			var combinedTest junit.Result
+			var cumFailure []string
+			for _, test := range tests {
+				// skipped test has no reason to rerun, so no deduplication
+				if test.Skipped != nil {
+					skipped = true
+					combinedTest = test
+				} else if test.Failure != nil {
+					if passed {
+						passed = false
+						failed = false
+						flaky = true
+					}
+					if !flaky {
+						failed = true
+					}
+					cumFailure = append(cumFailure, *test.Failure)
+					// Use the last failed result for flaky test, or same test
+					// failed multiple times
+					combinedTest = test
+				} else {
+					if failed {
+						passed = false
+						failed = false
+						flaky = true
+					}
+					if !flaky {
+						passed = true
+						// Don't use a succeeded test result for flaky test
+						combinedTest = test
+					}
+				}
+			}
+
+			if len(cumFailure) > 0 {
+				failure := strings.Join(cumFailure,
+					"\n\n---Separation line for tests that failed reruns---\n\n")
```

Comment:
This seems pretty hacky. IMO each should preserve the separate junit.Result like it does currently. It may be necessary/worthwhile to allow people to expand each flaky set of tests to the individual results.

Reply:
Totally agree that it's hacky, and it's certainly a good idea to allow people expand each flaky set of tests, since this causes no regression at all. I will think about how to make this work

Reply:
Katharine and michelle192837 should be able to point you in the right direction for the necessary typescript changes.

Reply:
Got plenty of help from them, and I think it's working now. This is almost to the best of my html knowledge